src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1750,22 +1750,21 @@ public final void stop() {
      * @revised 6.0, 14
      */
     public void interrupt() {
-        if (this != Thread.currentThread()) {
-            checkAccess();
-
-            // thread may be blocked in an I/O operation
-            synchronized (interruptLock) {
-                Interruptible b = nioBlocker;
-                if (b != null) {
-                    interrupted = true;
-                    interrupt0();  // inform VM of interrupt
-                    b.interrupt(this);
-                    return;
-                }
-            }
-        }
-        interrupted = true;
-        interrupt0();  // inform VM of interrupt
+		if (currentThread() != this) {
+			checkAccess();
+		}
+
+		synchronized (interruptLock) {
+			interrupt0();  // inform VM of interrupt
+			interrupted = true;
+
+			// thread may be blocked in an I/O operation
+			Interruptible b = nioBlocker;
+			if (b != null) {
+				b.interrupt(this);
+				return;
```

Comment:
nit: this return is redundant now.

Reply:
The PR should be valid - https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK19_ppc64le_linux/184/

Reply:
will change this once tests passes